client/src/elm/Pages/MessagingCenter/View.elm

```diff
@@ -66,74 +66,76 @@ view language currentTime nurseId nurse db model =
+                                                if surveyCount == 0 then
+                                                    -- We need to have at least one survey completed.
+                                                    True
+
+                                                else if surveyCount >= 3 then
+                                                    -- There can be up to 3 surveys during the program which lasts
+                                                    -- 6 months. At the beginning, after 3 months, and at the end.
+                                                    -- So, if we have 3 surveys already, there's no need for another one,
+                                                    False
+
+                                                else if diffMonthsProgramStartLastSurvey >= 6 then
+                                                    -- Last survey run after 6 months from the program start date.
+                                                    -- It means that the program has already ended there.
+                                                    -- No need to run another one.
+                                                    False
+
+                                                else if diffMonthsLastSurveyCurrent >= 3 then
+                                                    -- If we got so far, and the interval from the last survey
+                                                    -- is 3 months or more, we should run the survey.
+                                                    True
+
+                                                else
+                                                    False
+                                        in
+                                        List.head filteredSurveys
+                                            |> Maybe.map filterCondition
+                                            |> Maybe.withDefault True
 
-                    else if runAdoptionSurvey then
-                        viewAdoptionSurvey language currentDate nurseId model.surveyForm
+                                    runAdoptionSurvey =
+                                        runSurvey ResilienceSurveyAdoption
 
-                    else
-                        messagingCenterView
+                                    runQuarterlySurvey =
+                                        runSurvey ResilienceSurveyQuarterly
+                                in
+                                if runQuarterlySurvey then
+                                    viewQuarterlySurvey language currentDate nurseId model.surveyForm
+
+                                else if runAdoptionSurvey then
+                                    viewAdoptionSurvey language currentDate nurseId model.surveyForm
+
+                                else
+                                    messagingCenterView
+                            )
+                        |> Maybe.withDefault messagingCenterView
```

Comment:
@Diama1 This means that if nurse got no surveys filled, we will never ask to fill any survey.
So, this change is incorrect.